shared/fs/sync-status/container.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import {namedConnect} from '../../util/typed-connect'
+import SyncStatus from './index'
+
+type OwnProps = {|
+  path: Types.Path,
+|}
+
+const mapStateToProps = (state, ownProps) => ({
+  _kbfsDaemonStatus: state.fs.kbfsDaemonStatus,
+  _pathItem: state.fs.pathItems.get(ownProps.path, Constants.unknownPathItem),
+  _tlf: Constants.getTlfFromPath(state.fs.tlfs, ownProps.path),
+  _uploads: state.fs.uploads.syncingPaths,
+})
+
+const mapDispatchToProps = dispatch => ({})
+
+const mergeProps = (stateProps, dispatchProps, ownProps) => {
+  const status = Constants.getSyncStatusInMergeProps(
+    stateProps._kbfsDaemonStatus,
+    stateProps._tlf,
+    stateProps._pathItem,
+    stateProps._uploads,
+    ownProps.path
+  )
+  return {
+    folder: stateProps._pathItem.type === 'folder',
+    status: status,
+  }
```

Comment:
Seems like this could be in short-hand form by inlining `status`, but up to you!